### PR TITLE
Shrink docker image from 530MB -> 250MB

### PR DESCRIPTION
**Pull request by rmlearney-digicatapult**
# Pull Request

## Checklist
- [X] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [X] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [X] Chore

## High level description

Previous docker image was 530MB in size because the multi-stage build wasn't correctly omitting dev dependencies from the final build. 

## Detailed description

`npm ci --omit-dev` in the `service` stage of the `Dockerfile` didn't correctly omit dev dependencies before.

Have amended in order to correctly omit them at final build time with `npm ci --omit=dev`.

Also removed unnecessary installation of npm at each stage because this is already bundled within the `node:` images.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,9 +3,6 @@ FROM node:current-alpine AS builder
 
 WORKDIR /veritable-ui
 
-# Install base dependencies
-RUN npm install -g npm@10.x.x
```
Comment (mattdean-digicatapult) [inline Dockerfile:7]: we still want to install latest npm so please revert these

File: Dockerfile [hunk 2]
```diff
@@ -32,20 +27,17 @@ ENV NODE_ENV=${NODE_ENV}
...
 
 WORKDIR /veritable-ui
 
-RUN apk add --update coreutils curl 
-RUN npm -g install npm@10.x.x
+RUN apk add --no-cache coreutils curl
 
 COPY package*.json ./
-RUN npm ci --omit-dev
+RUN npm install --omit=dev
```
[the remark this answers is not in the document]
Comment (rmlearney-digicatapult) [reply Dockerfile:40]: `npm ci` will install all dev dependencies by following `package-lock` so the image will remain the same size. If you need all dependencies then this PR doesn't help

File: Dockerfile (hunk 1 above)
Comment (rmlearney-digicatapult) [reply Dockerfile:7]: Should I switch to `npm install -g npm@latest`?

Comment (rmlearney-digicatapult) [reply Dockerfile:7]: `npm update -g npm` adds 30MB
`npm install -g npm@latest` adds 40MB

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmlearney-digicatapult) [reply Dockerfile]: Done 👍

File: Dockerfile (hunk 1 above)
Comment (mattdean-digicatapult) [reply Dockerfile:7]: Leave it as `npm install -g npm@10.x.x` as this makes sure we're using the latest available `npm` which is matched by our `package.json` `engines` section. The extra image size is worth it. 

One additional change I might ask for though is to change the base image from `current-alpine` to `lts-alpine`? That would then also match our package json `engines`

Comment (rmlearney-digicatapult) [reply Dockerfile]: 290MB final image size

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**